websites/A/AlternativeScans/presence.ts

```diff
@@ -0,0 +1,115 @@
+
+presence.on('UpdateData', async () => {
+  const presenceData: PresenceData = {
+    largeImageKey: ActivityAssets.Logo,
+    startTimestamp: browsingTimestamp,
+    smallImageKey: Assets.Search,
+    type: ActivityType.Watching,
+  }
+  const { pathname, href } = document.location
+  const pathArr = pathname.split('/')
+  let pathred
+  let img
+  let titlename
+
+  if (pathArr[1]) {
+    pathred = pathArr[1].split('?')
+
+    switch (pathArr[1]) {
+      case 'series':
+        img = document.querySelector<HTMLImageElement>('.cover-column img.cover-image')
+
+        presenceData.largeImageKey = img?.src ?? ''
+        presenceData.smallImageKey = Assets.Viewing
+
+        titlename = document
+          .querySelector('head > title')
+          ?.textContent
+
+        presenceData.details = `Viewing:  ${titlename}`
+        presenceData.buttons = [{
+          label: 'View Series',
+          url: href,
+        }, {
+          label: 'Join Discord Server',
+          url: 'https://discord.gg/Rr5HWHuk7Y',
+        }]
+        break
+
+      case 'routes':
+        presenceData.details = `Searching for Series`
+        presenceData.buttons = [{
+          label: 'View Site',
+          url: 'https://alternativescans.icu/',
+        }, {
+          label: 'Join Discord Server',
+          url: 'https://discord.gg/Rr5HWHuk7Y',
+        }]
+
+        break
+
+      default:
+
+        if (pathred[0] === 'reader') {
+          const currentUrl = new URL(window.location.href)
+          const params = currentUrl.searchParams
+          const seriesparam = params.get('series')
+          const chapterparam = params.get('chapter')
+          let seriesparamup
+
+          if (seriesparam) {
+            seriesparamup = seriesparam.toUpperCase()
+          }
+
+          const name = document
+            .querySelector<HTMLAnchorElement>('#current-chapter-name a')
+            ?.textContent
+            ?.trim()
+            ?? ''
+
+          presenceData.details = `${name}`
+          presenceData.state = document
+            .querySelector('head > title')
+            ?.textContent
+            ?.replace(` - ${seriesparamup}`, '')
+          presenceData.largeImageKey = `https://files.alternativescans.icu/file/public-chapter-images/${seriesparam}/${chapterparam}/1.png`
+          presenceData.smallImageKey = Assets.Reading
+          presenceData.buttons = [{
+            label: 'Read Chapter',
+            url: href,
+          }, {
+            label: 'Join Discord Server',
+            url: 'https://discord.gg/Rr5HWHuk7Y',
+
+          }]
+        }
+    }
+  }
+  else {
+    presenceData.details = 'Viewing the Homepage'
+    presenceData.buttons = [{
+      label: 'View Site',
+      url: 'https://alternativescans.icu/',
+    }, {
+      label: 'Join Discord Server',
+      url: 'https://discord.gg/Rr5HWHuk7Y',
+    }]
+  }
+
+  if (presenceData.details) {
+    presence.setActivity(presenceData)
+  }
+  else {
+    presence.setActivity()
```

Comment:
If you want to completely hide the activity data rather than show a blank activity, use `presence.clearActivity()` instead